src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/DB2Engine.java

```diff
@@ -67,32 +70,32 @@
     /**
      * The DB2 JDBC driver.
      */
-    protected static final String DB2_DRIVER = DatabaseEngineDriver.DB2.driver();
+    private static final String DB2_DRIVER = DatabaseEngineDriver.DB2.driver();
+
     /**
      * Name is already used by an existing object.
      */
-    public static final String NAME_ALREADY_EXISTS = "DB2 SQL Error: SQLCODE=-601, SQLSTATE=42710";
+    private static final int NAME_ALREADY_EXISTS = -601;
+
     /**
      * Table can have only one primary key.
      */
-    public static final String TABLE_CAN_ONLY_HAVE_ONE_PRIMARY_KEY = "DB2 SQL Error: SQLCODE=-624, SQLSTATE=42889";
-    /**
-     * Sequence does not exist.
-     */
-    public static final String SEQUENCE_DOES_NOT_EXIST = "DB2 SQL Error: SQLCODE=-204, SQLSTATE=42704";
+    private static final int TABLE_CAN_ONLY_HAVE_ONE_PRIMARY_KEY = -624;
+
     /**
-     * Table or view does not exist.
+     * The name (e.g. of a sequence, table, view, ...) does not exist.
      */
-    public static final String TABLE_OR_VIEW_DOES_NOT_EXIST = "DB2 SQL Error: SQLCODE=-204, SQLSTATE=42704";
+    private static final int NAME_DOES_NOT_EXIST = -204;
+
     /**
-     * Foreign key already exists
+     * The default size of a BLOB in DB2.
      */
-    public static final String FOREIGN_ALREADY_EXISTS = "DB2 SQL Error: SQLCODE=-601, SQLSTATE=42710";
+    static final String DB2_DEFAULT_BLOB_SIZE = "2G";
 
     /**
-     * The default size of a BLOB in DB2.
+     * The column must have a {@link DbColumnConstraint#NOT_NULL NOT NULL column constraint}.
      */
-    public static final String DB2_DEFAULT_BLOB_SIZE = "2G";
+    private static final int COLUMN_MUST_BE_NON_NULL = -542;
```

Comment:
I'd say that what makes the public API in PDB is everything except DB-specific code:
interfaces, enums, exceptions, and all dml and ddl classes that are not some DB implementation.  
Maybe one day we can separate this public stuff from the rest into separate modules, and do breaking changes checks :smile: 

This is an implementation for DB2, and not only we should rely on the SQL code instead of error messages too check what type of error is occurring on the DB, but also there's no reason for these codes to be public.

Reply:
> This is an implementation for DB2, and not only we should rely on the SQL code instead of error messages too check what type of error is occurring on the DB, but also there's no reason for these codes to be public.

Completly agree, this way is so much better than before, I was just worried that we would break something.